src/sql/src/query_model/hir.rs

```diff
@@ -129,6 +129,88 @@ impl FromHir {
                 }
                 box_id
             }
+            HirRelationExpr::Distinct { input } => {
+                let select_id = self.generate_select(*input);
+                let mut select_box = self.model.get_mut_box(select_id);
+                select_box.distinct = DistinctOperation::Enforce;
+                select_id
+            }
+            HirRelationExpr::Reduce {
+                input,
+                group_key,
+                aggregates,
+                expected_group_size: _,
+            } => {
+                // An intermediate Select Box is generated between the input box and
+                // the resulting grouping box so that the grouping key and the arguments
+                // of the aggregations contain only column references
+                let input_box_id = self.generate_internal(*input);
+                let select_id = self.model.make_select_box();
+                let input_q_id =
+                    self.model
+                        .make_quantifier(QuantifierType::Foreach, input_box_id, select_id);
+                let group_box_id = self
+                    .model
+                    .make_box(BoxType::Grouping(Grouping { key: Vec::new() }));
+                let select_q_id =
+                    self.model
+                        .make_quantifier(QuantifierType::Foreach, select_id, group_box_id);
+                let mut key = Vec::new();
+                for k in group_key.into_iter() {
+                    // Make sure the input select box projects the source column needed
+                    let col_ref = BoxScalarExpr::ColumnReference(ColumnReference {
```

Comment:
Can this `col_ref` and the later `col_ref` get slightly different names? Like `input_col_ref` and `select_box_col_ref`?

Reply:
Good point. Done